fem/prestriction.cpp

```diff
@@ -329,7 +325,7 @@ void ParL2FaceRestriction::DoubleValuedConformingMult(
    auto d_indices1 = scatter_indices1.Read();
    auto d_indices2 = scatter_indices2.Read();
    auto d_x = Reshape(x.Read(), t?vd:ndofs, t?ndofs:vd);
-   auto d_x_shared = Reshape(x_gf.FaceNbrData().Read(),
+   auto d_x_shared = Reshape(face_nbr_data.Read(),
```

Comment:
Since `face_nbr_data` is an alias (when this method is called with a `ParGridFunction` argument `x`) it will be good to sync it to its base to avoid repeated host -> device transfers. The same should be done in the other places where `GetLVectorFaceNbrData` is used. The sync can be done right here after the call to `Read()` -- one needs to get a pointer to the base `Vector` (or `NULL` when `face_nbr_data` is not an alias) from `GetLVectorFaceNbrData` to be able to do the sync easily.

Reply:
A simpler solution maybe to call `.Read()` for the base `Vector` in `GetLVectorFaceNbrData` before creating the alias.

Reply:
OK, I did this in 94906f661 if I understood your suggestion correctly.

Reply:
Yes, that's what I was thinking.